src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -1275,6 +1277,39 @@ private boolean insideAnObjectCreation(MethodDeclaration decl) {
     throw new RuntimeException("Got a method declaration with no class!");
   }
 
+  /**
+   * Given a type variable, update the list of synthetic classes accordingly. Node: while the type
+   * of the input for this method is ClassOrInterfaceType, it is actually a type variable. Make sure
+   * to check with {@link UnsolvedSymbolVisitor#isTypeVar(String)} before calling this method.
+   *
+   * @param type a type variable to be used as input.
+   */
+  private void updateSyntheticClassesForTypeVar(ClassOrInterfaceType type) {
+    String typeSimpleName = type.getNameAsString();
+    for (Map<String, NodeList<ClassOrInterfaceType>> typeScope : typeVariables) {
+      if (typeScope.containsKey(typeSimpleName)) {
+        NodeList<ClassOrInterfaceType> boundOfType = typeScope.get(typeSimpleName);
+        for (int index = 0; index < boundOfType.size(); index++) {
+          try {
+            boundOfType.get(index).resolve();
+          } catch (UnsolvedSymbolException | UnsupportedOperationException e) {
+            if (e instanceof UnsolvedSymbolException) {
+              // quoted from the documentation of Oracle: "A type variable with multiple bounds is a
+              // subtype of all the types listed in the bound. If one of the bounds is a class, it
+              // must be specified first."
+              // If the first bound is also unsolved, it is better to assume it to be a class.
+              if (index == 0) {
+                solveSymbolsForClassOrInterfaceType(boundOfType.get(index), false);
+              } else {
+                solveSymbolsForClassOrInterfaceType(boundOfType.get(index), true);
```

Comment:
Rather than an if-statement with two calls here, I would find this easier to read if you defined a boolean local variable for whether or not we're going to create an interface (based on the index).

Reply:
Should there also be a call to `gotException()` here?